python/feathub/registries/mysql_registry.py

```diff
@@ -0,0 +1,273 @@
+        self.table = mysql_registry_config.get(TABLE_CONFIG)
+        self.host = mysql_registry_config.get(HOST_CONFIG)
+        self.port = mysql_registry_config.get(PORT_CONFIG)
+        self.username = mysql_registry_config.get(USERNAME_CONFIG)
+        self.password = mysql_registry_config.get(PASSWORD_CONFIG)
+
+        # dict that acts as a local cache for built and registered tables.
+        # each value in the dict is a tuple of table before build and table after build.
+        self.tables: Dict[str, Tuple[TableDescriptor, TableDescriptor]] = {}
+
+        self.conn = mysql.connector.connect(
+            host=self.host,
+            port=self.port,
+            user=self.username,
+            password=self.password,
+            database=self.database,
+        )
+        self.cursor = self.conn.cursor()
+
+        self.cursor.execute(
+            f"""
+                CREATE TABLE IF NOT EXISTS `{self.table}`(
+                   `digest` VARCHAR(64) NOT NULL,
+                   `name` TEXT NOT NULL,
+                   `timestamp` TIMESTAMP NOT NULL,
+                   `is_deleted` BOOLEAN NOT NULL,
+                   `original_json` TEXT NOT NULL,
+                   `built_json` TEXT NOT NULL,
+                   PRIMARY KEY ( `digest`, `timestamp` )
+                );
+            """
+        )
+
+    def build_features(
+        self,
+        feature_descriptors: List[TableDescriptor],
+        force_update: bool = False,
+        props: Optional[Dict] = None,
+    ) -> List[TableDescriptor]:
+        result = []
+        for table in feature_descriptors:
+            if table.name == "":
+                raise FeathubException(
+                    "Cannot build a TableDescriptor with empty name."
+                )
+            self.tables[table.name] = (
+                table,
+                table.build(self, force_update=force_update, props=props),
+            )
+            result.append(self.tables[table.name][1])
+
+        return result
+
+    def register_features(
+        self, feature_descriptors: List[TableDescriptor], force_update: bool = False
+    ) -> bool:
+        self.build_features(feature_descriptors, force_update=force_update)
+
+        for descriptor in feature_descriptors:
+            existing_features = self._get_feature_from_remote_registry_or_none(
+                descriptor.name
+            )
+            if (
+                existing_features is not None
+                and existing_features[1].to_json()
+                == self.tables[descriptor.name][1].to_json()
+            ):
+                continue
+
+            original_json = json.dumps(
+                self.tables[descriptor.name][0].to_json(), sort_keys=True
+            ).replace('"', '\\"')
+            built_json = json.dumps(
+                self.tables[descriptor.name][1].to_json(), sort_keys=True
+            )
+            digest = _get_digest(built_json)
```

Comment:
Should digest also depend on the `original_descriptor`?

And would it be simpler to calculate digest after `built_json.replace('"', '\\"')`?